kedro-datasets/test_requirements.txt

```
@@ -24,6 +24,7 @@ lxml~=4.6
 matplotlib>=3.0.3, <3.4; python_version < '3.10' # 3.4.0 breaks holoviews
 matplotlib>=3.5, <3.6; python_version == '3.10'
 memory_profiler>=0.50.0, <1.0
+mlflow==2.2.1
```

```suggestion
mlflow~=2.2.1; python_version>='3.8'
mlflow~=1.30.0; python_version=='3.7'
```

This dataset needs to support Python 3.7. [`mlflow 1.30.0` was the last version of `mlflow` to support 3.7](https://github.com/mlflow/mlflow/releases/tag/v2.0.1).

I would also use a compatible release operator here, not `==`.